services/merak-topo/handler/handler.go

```diff
@@ -242,174 +330,326 @@ func UpdateComputenodeInfo(client *kubernetes.Clientset, topo_id string, returnM
 				// }
 
 				cnode.Mac = "ff:ff:ff:ff:ff:ff"
-				cnode.OperationType = pb_common.OperationType_INFO
+				cnode.OperationType = database.OPERATION_INFO
 
 				if len(res.Status.ContainerStatuses) == 0 {
+					cnode.Status = database.STATUS_NONE
 					log.Printf("updatecomputenode: container status is not available %v", res.Name)
 				} else {
 					if res.Status.ContainerStatuses[len(res.Status.ContainerStatuses)-1].Ready {
-						cnode.Status = pb_common.Status_READY
-					} else {
-						cnode.Status = pb_common.Status_DEPLOYING
+						cnode.Status = database.STATUS_READY
 					}
+
 				}
-				returnMessage.ComputeNodes = append(returnMessage.ComputeNodes, &cnode)
-			}
-		}
-	}
 
-	log.Printf("updatecomputenode:=========Update compute nodes information ===========")
+				err_db := database.SetValue(topo_id+":"+node.Name, cnode)
+				if err_db != nil {
+					return fmt.Errorf("updatecomputenode: save compute node error %s", err_db.Error())
+				}
 
-	err_db2 := database.SetPbReturnValue(topo_id+":updateReturnmsg", returnMessage)
-	if err_db2 != nil {
-		log.Printf("fail to save k8s cluster host node to DB %s", err_db2)
-		return fmt.Errorf("fail to save k8s cluster host node to DB %s", err_db2)
-	}
-	return nil
-}
+			}
 
-func Info(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.ReturnTopologyMessage) error {
+		}
 
-	err := UpdateComputenodeInfo(k8client, topo_id, returnMessage)
-	if err != nil {
-		return fmt.Errorf("fail to update compute nodes info %s", err)
 	}
 
-	log.Printf("after updating compute node info: %s", returnMessage)
+	elaps := time.Since(start_time0)
 
-	err_check2 := database.GetPbReturnValue(topo_id+":updateReturnmsg", "", returnMessage)
-	if err_check2 != nil {
-		return fmt.Errorf("fail to find return message compute nodes info %s", err_check2)
-	}
+	log.Printf("updatecomputenode:=========complete updating compute nodes info in DB =========== %v", elaps)
 
 	return nil
 }
 
-func Delete(k8client *kubernetes.Clientset, topo_id string) error {
-	topo, err_db := database.FindTopoEntity(topo_id, "")
-
-	if err_db != nil {
-		return fmt.Errorf("query topology_id error %s", err_db)
-	}
-
-	err := Topo_delete(k8client, topo)
-	if err != nil {
-		return fmt.Errorf("topology delete fails %s", err)
-	}
-
-	return nil
-}
+func Info(k8client *kubernetes.Clientset, topo_id string, returnMessage *pb.ReturnTopologyMessage) error {
 
-func QueryMac(k8client *kubernetes.Clientset, topo_id string) error {
+	// err := UpdateComputenodeInfo(k8client, topo_id)
+	// if err != nil {
+	// 	return fmt.Errorf("fail to update compute nodes info %s", err)
+	// }
```

Comment:
@zzengcs if you don't need it, please remove it.

Reply:
Resolved